docs/deprecation_policy.rst

```diff
@@ -1,3 +1,105 @@
+version which add new features and bug fixes. For the most recent minor version
+there will also be new patch versions published as bugs are identified and fixed
+on that release series.
+
+For the purposes of semantic versioning, the Qiskit public API is considered
+any documented module, class, function, or method that is not marked as private
+(with a ``_`` prefix). The supported Python versions, minimum supported Rust
+version (for building Qiskit from source), and any Python package dependencies
+(including the minimum supported versions of dependencies) used by Qiskit are
+not part of the backwards compatibility guarantees and may change during any
+release. Only minor or major version releases will raise minimum requirements
+for using or building Qiskit (including adding new dependencies), but patch
+fixes might include support for new versions of Python or other dependencies.
+Usually the minimum version of a dependency is only increased when older
+dependency versions go out of support or when it is not possible to maintain
+compatibility with the latest release of the dependency and the older version.
+
+<TODO Add calendar diagrams>
+
+With the release of a new major version, the previous major version is supported
+for at least 6 months; only bug and security fixes will be accepted during this
+time and only patch releases will be published for this major version. A final
+patch version will be published when support is dropped and that release will
+also document the end of support for that major version series. A longer
+support window is needed for the previous major version as this gives downstream
+consumers of Qiskit a chance to migrate not only their code but also their
+users. It's typically not recommended for a downstream library that
+depends on Qiskit to bump its minimum Qiskit version to a new
+major version release immediately because its user base also needs a chance
+to migrate to the new API changes. Having an extended support window
+for the previous major Qiskit version gives downstream projects time to fix
+compatibility with the next major version. Downstream projects can provide support for two
+release series at a time to give their users a migration path.
+
+Upgrade Strategy
+================
+
+Whenever a new major version is released the recommended upgrade path
+is to first upgrade to use the most recent minor version on the previous major
+version. Immediately preceding a new major version a final minor version will
+be published. This final minor version release ``M.N+1.0`` is equivalent to
+``M.N.0`` but with warnings and deprecations for any API changes that are
+made on the new major version series.
+
+For example, on the release of Qiskit 1.0.0 a 0.46.0 release was published
+immediately proceeding the 1.0.0 release. The 0.46.0 release was equivalent
+to the 0.45.0 release but with additional deprecation warnings that document
```

Comment:
You can add deprecations in any minor/major version, the exception being 1.0, because we're trying to make 1.0 deprecation warning free. For example I'm already planning deprecations for 1.1: https://github.com/Qiskit/qiskit/issues/11212